include/graphblas/reference/io.hpp

```diff
@@ -1156,6 +1156,206 @@ namespace grb {
 		}
 	}
 
+	template<
+		Descriptor descr = descriptors::no_operation,
+		typename OutputType, typename MaskType, typename InputType,
+		typename RIT1, typename CIT1, typename NIT1,
+		typename RIT2, typename CIT2, typename NIT2
+	>
+	RC set(
+		Matrix< OutputType, reference, RIT1, CIT1, NIT1 > &C,
+		const Matrix< MaskType, reference, RIT2, CIT2, NIT2 > &Mask,
+		const Matrix< InputType, reference, RIT2, CIT2, NIT2 > &A,
+		const Phase &phase = EXECUTE
+	) noexcept {
+		static_assert( !std::is_void< OutputType >::value,
+			"grb::set (masked set to matrix): cannot have a pattern "
+			"matrix as output" );
+		static_assert( std::is_convertible< InputType, OutputType >::value,
+			"grb::set (masked set to matrix): input type cannot be "
+			"converted to output type"
+		);
+		static_assert(
+			! ( descr & descriptors::structural && descr & descriptors::invert_mask ),
+			"grb::set can not be called with both descriptors::structural "
+			"and descriptors::invert_mask in the masked variant"
+		);
+#ifdef _DEBUG
+		std::cout << "Called grb::set (matrix-to-matrix-masked, reference)\n";
+#endif
+		// static checks
+		NO_CAST_ASSERT( ( !(descr & descriptors::no_casting) ||
+				std::is_same< InputType, OutputType >::value
+			), "grb::set",
+			"called with non-matching value types"
+		);
+
+		// dynamic checks
+		assert( phase != TRY );
+
+		const size_t nrows = grb::nrows( C );
+		const size_t ncols = grb::ncols( C );
+
+		const size_t m = grb::nrows( Mask );
+		const size_t n = grb::ncols( Mask );
+
+		/*grb::Monoid<
+			grb::operators::left_assign< OutputType >,
+			grb::identities::zero
+		> dummyMonoid;*/
+
+		if( m == 0 || n == 0 ) {
+			// If the mask has a null size, it will be ignored
+			return set< descr >( C, A, phase );
+		}
+
+		if( nrows != grb::nrows( A ) || nrows != m ) {
+			return MISMATCH;
+		}
+
+		if( ncols != grb::ncols( A ) || ncols != n ) {
+			return MISMATCH;
+		}
+
+		const auto &mask_raw = internal::getCRS( Mask );
+
+		const auto &A_raw = internal::getCRS( A );
+
+		size_t nzc = 0;
+
+		char * mask_arr = nullptr;
+		char * mask_buf = nullptr;
+		OutputType * mask_valbuf = nullptr;
+		internal::getMatrixBuffers( mask_arr, mask_buf, mask_valbuf, 1, Mask );
+
+		internal::Coordinates< reference > mask_coors;
+		mask_coors.set( mask_arr, false, mask_buf, ncols );
+
+		for( size_t i = 0; i < nrows; ++i ) {
+			mask_coors.clear();
+			for( auto k = mask_raw.col_start[ i ]; k < mask_raw.col_start[ i + 1 ]; ++k ) {
+				const size_t k_col = mask_raw.row_index[ k ];
```

Comment:
```suggestion
				const auto k_col = mask_raw.row_index[ k ];
```
Since we use `RTI`, `CIT`, etc.